cachers/counts.go

```diff
@@ -0,0 +1,125 @@
+package cachers
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"sync/atomic"
+)
+
+// Counts keeps counts cache events
+type Counts struct {
+	gets      atomic.Int64
+	hits      atomic.Int64
+	misses    atomic.Int64
+	puts      atomic.Int64
+	getErrors atomic.Int64
+	putErrors atomic.Int64
+}
+
+func (c *Counts) Summary() string {
+	return fmt.Sprintf("%d gets (%d hits, %d misses, %d errors); %d puts (%d errors)",
+		c.gets.Load(), c.hits.Load(), c.misses.Load(), c.getErrors.Load(), c.puts.Load(), c.putErrors.Load())
+}
+
+type LocalCacheWithCounts struct {
+	Counts
+	cache LocalCache
+}
+
+func (l *LocalCacheWithCounts) Kind() string {
+	return l.Kind()
```

Comment:
```suggestion
	return l.cache.Kind()
```

Or it will recurse indefinitely